### PR TITLE
fix(jtk): replace string-based 404 detection in move/type-change

### LINKED ISSUE #280: fix(jtk): move and type-change commands misreport success as "not available on Jira Cloud" on post-operation 404s
## Summary

`jtk issues move` and `jtk issues update --type` both use `strings.Contains` to detect whether a 404 means \"Jira Server/DC\" — but this is too broad. Any 404 that occurs after the move has already been dispatched (e.g., polling the task status for an issue key that has changed) trips the same branch and prints a misleading error, even though the operation succeeded.

## Affected code paths

| File | Line | Command |
|------|------|---------|
| `tools/jtk/internal/cmd/issues/move.go:153` | `MoveIssues()` error check | `jtk issues move` |
| `tools/jtk/internal/cmd/issues/update.go:228` | `MoveIssues()` error check | `jtk issues update --type` |

Both blocks look like:

```go
if strings.Contains(err.Error(), "404") || strings.Contains(err.Error(), "not found") {
    return fmt.Errorf("... this feature is only available on Jira Cloud")
}
```

## Repro

1. Create an SDLC issue in the MON project (e.g. `jtk issues create -p MON -t SDLC -s "test"`)
2. Move it to the INT project:
   ```
   jtk issues move MON-XXXX --to-project INT
   ```
3. Observe output:
   ```
   Moving 1 issue(s) to INT (SDLC)...
   move operation failed - this feature is only available on Jira Cloud
   ```
4. Run `jtk issues get MON-XXXX` — the issue has actually been moved (now shows as INT-YYY), confirming the move succeeded and the error is spurious.

The same false-failure is reachable via `jtk issues update MON-XXXX --type <type>`, which calls the same `MoveIssues` API path and has identical error-handling logic.

## Root cause

The bulk move API is asynchronous. `MoveIssues` POSTs and returns a `taskId`. The 404 is most likely occurring during the subsequent `GetMoveTaskStatus` poll — either because the issue key changed mid-poll or the task endpoint returned 404 transiently — but the string-match catches it and surfaces it as an unsupported-platform error rather than propagating it or treating it as completion.

Additionally, `strings.Contains(err.Error(), "not found")` is fragile: `ErrNotFound` from `shared/errors` wraps the sentinel with `errors.Is`-compatible chaining, so the correct check is already available.

## Fix direction

Replace string matching with `errors.Is`:

```go
// shared/errors already provides this
if errors.Is(err, api.ErrNotFound) {
    return fmt.Errorf("... this feature is only available on Jira Cloud")
}
```

This limits the "not on Cloud" message to genuine endpoint-not-found responses on the initial POST, rather than any 404 that appears anywhere in the call chain. The task status poll loop should separately handle 404s (potentially as a completion signal or a retryable transient).

## Success criteria

- `jtk issues move MON-XXXX --to-project INT` exits 0 and prints `Moved 1 issue(s) to INT` when the move completes successfully — no spurious "only available on Jira Cloud" error.
- `jtk issues get MON-XXXX` after a successful move resolves to the new INT-YYY key (unchanged by this fix, but confirms the move actually happened).
- A genuine Server/DC rejection (HTTP 404 on the initial `POST /bulk/issues/move`) still surfaces the "only available on Jira Cloud" message.
- `jtk issues update MON-XXXX --type <type>` similarly exits 0 and prints a success confirmation when the type change completes.

## Integration test

Add a case to `integration-tests.md` covering cross-project move: create a throwaway SDLC issue in MON, move it to INT, verify the command exits 0 and the issue is retrievable under its new INT key. This would have caught the repro above.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

- Replace `strings.Contains(err.Error(), "404")` with `sharederrors.IsNotFound(err)` in `issues move` and `issues update --type`
- Extract shared `pollMoveTask` helper with retry logic for transient 404s during task status polling
- Degrade gracefully on persistent poll 404s (exit 0 with advisory instead of false failure)
- `update --type` returns `errTypeChangeUnverified` so remaining field updates proceed normally

## Test plan

- [x] `make build` — compiles
- [x] `make test` — all tests pass (1729 jtk + 948 cfl)
- [x] `make lint` — 0 issues
- [x] New `poll_test.go` covers: terminal states, transient 404 retry, persistent 404 sentinel, context cancellation
- [ ] Integration: `jtk issues move MON-XXXX --to-project INT` exits 0 on success

Closes #280

### CODE REVIEW

**Comment by rianjs** [comment]
## Findings

### Major
1. The PR changes the command-level success/fallback behavior in [move.go](/Users/rianjs/dev/atlassian-cli/tools/jtk/internal/cmd/issues/move.go:170) and [update.go](/Users/rianjs/dev/atlassian-cli/tools/jtk/internal/cmd/issues/update.go:86), but only adds `pollMoveTask` unit tests. Please add command tests for persistent poll 404: `runMove` should exit nil with initiated stdout + advisory stderr, and `runUpdate --type ... --summary ...` should emit the unverified advisory while still applying the field update. That is the product behavior this ticket is meant to protect.

### Minor
1. [poll_test.go](/Users/rianjs/dev/atlassian-cli/tools/jtk/internal/cmd/issues/poll_test.go:98) mutates package global `pollRetryDelay` inside a `t.Parallel()` test while other parallel tests also depend on that variable. This can race or make the retry tests unexpectedly sleep for seconds. Make the delay override serial, restore per-test without `t.Parallel`, or pass the delay into the helper/test harness.

**Comment by rianjs** [comment]
## Findings

No findings. The added command-level tests now cover both `runMove` graceful degradation and `runUpdate --type` continuing field updates after an unverified type-change task. The poll delay override is no longer mutated from a parallel test.

**Comment by rianjs** [comment]
## TDD Coverage Assessment

### poll_test.go — adequate with one gap

Covers: terminal COMPLETE, terminal FAILED, transient 404 → success (call-count verified), persistent 404 → `errStatusUnavailable`, context cancellation during retry delay.

**Gap**: No test for context cancellation during the ENQUEUED/RUNNING sleep path. The code has a `select { case <-ctx.Done() }` there (line 51–53 of poll.go), but all cancellation tests go through the 404 retry path. Low risk since the select is structurally identical, but the branch is untested.

**Minor**: CANCELLED terminal state is handled in poll.go's switch but has no dedicated poll-level test (only tested indirectly via move_test.go's terminal cases). Similarly for the `default:` branch (unknown status returned without error).

---

### move_test.go — adequate

Covers:
- Graceful degradation when poll returns persistent 404 (`TestRunMove_PollNotFound_GracefulDegradation`): exit 0, task ID in stdout, advisory in stderr.
- Server/DC detection when the *initiate* endpoint 404s (`TestRunMove_MoveIssuesNotFound_ServerDCError`): error contains "only available on Jira Cloud".

**Gap**: The graceful degradation test only asserts `"status unavailable"` appears in stderr. The move.go code also prints "verify with `jtk issues get`" — that substring isn't asserted. Trivial, but worth noting.

---

### update_test.go — one meaningful gap

Covers:
- Successful type change end-to-end (`TestRunUpdate_TypeChange`).
- Poll 404 → `errTypeChangeUnverified` → stderr warning + field updates continue (`TestRunUpdate_TypeChange_PollNotFound_ContinuesFieldUpdates`): PUT called, stderr contains "could not be verified".

**Gap**: No test for the Server/DC detection path in `changeIssueType` — i.e., when the *bulk/issues/move* POST itself returns 404. `move.go` has `TestRunMove_MoveIssuesNotFound_ServerDCError` for this, but the parallel path in `update.go` (line ~249: `"type change failed - this feature requires Jira Cloud"`) has no coverage. This is the original bug's detection logic and deserves a test.

---

### Summary

| Area | Status |
|---|---|
| poll.go core logic | Good |
| Context cancel during ENQUEUED/RUNNING | Missing |
| move graceful degradation | Good |
| move Server/DC detection | Good |
| update type change success | Good |
| update poll 404 → field updates continue | Good |
| update Server/DC detection (move POST 404) | **Missing** |

The update Server/DC gap is the most meaningful — it's the exact scenario the PR fixes and it has no test in update_test.go.

File: tools/jtk/internal/cmd/issues/poll.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+				case <-time.After(pollRetryDelay):
+				}
+				continue
+			}
+			if sharederrors.IsNotFound(err) {
+				return nil, errStatusUnavailable
+			}
+			return nil, err
+		}
+		notFoundRetries = 0
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/poll.go:44]: 🟡 **Medium** (harness-engineering:harness-architecture-reviewer): notFoundRetries is reset to 0 on every successful response, making maxPollNotFoundRetries=3 apply only to consecutive 404s. A sustained alternating success/404 pattern would retry indefinitely. The counter should accumulate across the poll lifetime, or the reset behavior should be explicitly documented as intentional.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/issues/poll.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+		switch status.Status {
+		case "COMPLETE", "FAILED", "CANCELLED":
+			return status, nil
+		case "ENQUEUED", "RUNNING":
+			select {
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			case <-time.After(1 * time.Second):
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/poll.go:53]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): The default case returns (status, nil) for unknown status strings rather than an error, creating an implicit caller contract where both move.go and update.go must each reproduce a default error branch. pollMoveTask should return an error directly for unknown statuses to make the contract explicit.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/issues/poll.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	sharederrors "github.com/open-cli-collective/atlassian-go/errors"
+
+	"github.com/open-cli-collective/jira-ticket-cli/api"
+)
+
+var errStatusUnavailable = errors.New("task status unavailable after retries")
+
+var errTypeChangeUnverified = errors.New("type change accepted but status could not be verified")
+
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/poll.go:16]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): errTypeChangeUnverified is defined in poll.go but is semantically owned by the update/type-change flow in update.go. Moving it to update.go would clarify ownership and avoid readers of poll.go needing to understand update-specific semantics.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/issues/poll_test.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	"time"
+
+	"github.com/open-cli-collective/atlassian-go/testutil"
+
+	"github.com/open-cli-collective/jira-ticket-cli/api"
+)
+
+func init() {
+	pollRetryDelay = 0
+}
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/poll_test.go:20]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): The package-level init() sets pollRetryDelay=0 for all tests in the package. TestPollMoveTask_ContextCancellation (non-parallel) temporarily mutates it to 5s. Running with -race or -count=2 may expose a data race. Prefer injecting delay as a parameter to pollMoveTask rather than relying on a mutable package-level variable.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/issues/update.go [hunk 5]
```diff
@@ -83,7 +84,11 @@ func runUpdate(ctx context.Context, opts *root.Options, issueKey, summary, descr
 	// Handle type change via the move API
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/update.go:84]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): When errTypeChangeUnverified is returned, runUpdate proceeds to apply field updates (PUT) while the type-change move task may still be in flight. If the task completes after the PUT, written field values could conflict with post-migration state. At minimum a comment explaining the known race and recovery steps would reduce confusion.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `0ee2ce6`

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 3 |
| harness-engineering:harness-enforcement-reviewer | 1 |
| harness-engineering:harness-knowledge-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (3 findings)</strong></summary>

**⚠️ Should Fix** - `tools/jtk/internal/cmd/issues/poll.go:44`
> notFoundRetries is reset to 0 on every successful response, making maxPollNotFoundRetries=3 apply only to consecutive 404s. A sustained alternating success/404 pattern would retry indefinitely. The counter should accumulate across the poll lifetime, or the reset behavior should be explicitly documented as intentional.

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/poll.go:53`
> The default case returns (status, nil) for unknown status strings rather than an error, creating an implicit caller contract where both move.go and update.go must each reproduce a default error branch. pollMoveTask should return an error directly for unknown statuses to make the contract explicit.

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/poll_test.go:20`
> The package-level init() sets pollRetryDelay=0 for all tests in the package. TestPollMoveTask_ContextCancellation (non-parallel) temporarily mutates it to 5s. Running with -race or -count=2 may expose a data race. Prefer injecting delay as a parameter to pollMoveTask rather than relying on a mutable package-level variable.

</details>

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/poll.go:16`
> errTypeChangeUnverified is defined in poll.go but is semantically owned by the update/type-change flow in update.go. Moving it to update.go would clarify ownership and avoid readers of poll.go needing to understand update-specific semantics.

</details>

<details>
<summary><strong>harness-engineering:harness-knowledge-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/update.go:84`
> When errTypeChangeUnverified is returned, runUpdate proceeds to apply field updates (PUT) while the type-change move task may still be in flight. If the task completes after the PUT, written field values could conflict with post-migration state. At minimum a comment explaining the known race and recovery steps would reduce confusion.

</details>

*2 info-level observations excluded. Run with `--verbose` to include.*

*2 PR discussion threads considered.*

---
<details>
<summary>Completed in 2m 22s | $0.83</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 22s wall · 7m 08s compute (Reviewers: 1m 46s · Synthesis: 33s) |
| Cost | $0.83 |
| Tokens | 221.8k in / 27.8k out |
| Turns | 16 |

</details>

File: tools/jtk/internal/cmd/issues/poll.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/poll.go:44]: Intentional design: resetting on success means 404s between successful responses are treated as transient (the endpoint is intermittently reachable). Accumulating across the poll lifetime would penalize healthy retry scenarios where one 404 occurs early, then the endpoint stabilizes. The 3-retry cap only fires on *consecutive* failures, which is the right signal for 'this endpoint is genuinely gone.'

File: tools/jtk/internal/cmd/issues/poll.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/poll.go:53]: This is intentional — both callers already have a default error branch for unknown statuses. Returning the status lets callers include the status string in their error messages, which is more informative than a generic error from the poll helper.

File: tools/jtk/internal/cmd/issues/poll.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/poll.go:16]: Acknowledged. It's in poll.go because both sentinels (errStatusUnavailable and errTypeChangeUnverified) represent poll-outcome states. Moving it would split related concerns. Keeping it here makes the poll outcome model self-contained.

File: tools/jtk/internal/cmd/issues/poll_test.go (hunk 4 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/poll_test.go:20]: The init() approach is simple and avoids adding a parameter to pollMoveTask that production code never varies. The context cancellation test is serial (not parallel) specifically to avoid races. -count=2 won't race because init() runs once per process.

File: tools/jtk/internal/cmd/issues/update.go (hunk 5 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/update.go:84]: This race exists in the happy path too — the type change is always asynchronous. The errTypeChangeUnverified case is actually less risky: the move task was accepted, and the field update is a standard PUT that Jira applies atomically. Adding a comment about the known race window.

**Comment by rianjs** [comment]
These are low-value, please approve the PR

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `4c1118a`

### Human Override

Approved at the request of @rianjs ([comment](https://github.com/open-cli-collective/atlassian-cli/pull/334#issuecomment-4363671219)).

> These are low-value, please approve the PR

No automated reviewers were run.
